CMakeLists.txt

```diff
@@ -524,11 +524,11 @@ IF (ZLIB_FOUND)
 ENDIF (ZLIB_FOUND)
 
 IF (WSLAY_FOUND)
-    ADD_LIBRARY(libh2o lib/websocket.c ${LIB_SOURCE_FILES})
-    ADD_LIBRARY(libh2o-evloop lib/websocket.c ${LIB_SOURCE_FILES})
+    ADD_LIBRARY(libh2o lib/websocket.c ${LIB_SOURCE_FILES} ${BROTLI_SOURCE_FILES})
+    ADD_LIBRARY(libh2o-evloop lib/websocket.c ${LIB_SOURCE_FILES} ${BROTLI_SOURCE_FILES})
```

Comment:
Would it make more sense to add brotli to libh2o and libh2o-evloop using `TARGET_LINK_LIBRARIES`?

I think it is better to refer to them as libraries (like we do for libuv), than embeding the brotli code in libh2o.

Reply:
I got it. I have changed at 0efe562

Reply:
linking against brotli if it is available on the host is the better option probably.  

@syohex your latests change works when brotli and brotli-dev installed.  Does **not** work when not installed - see below.  When not installed, it could fall back to your earlier approach and embed the code.  

```
$ cmake -B build . -DCMAKE_BUILD_TYPE=None -DBUILD_SHARED_LIBS=ON -DCMAKE_INSTALL_LIBDIR=/usr/lib -DCMAKE_INSTALL_PREFIX=/usr -DWITH_MRUBY
=ON
-- The C compiler identification is GNU 9.2.0
-- The CXX compiler identification is GNU 9.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.6.3") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found OpenSSL: /usr/lib/libcrypto.so (found version "1.1.1g")  
-- Found ZLIB: /lib/libz.so (found version "1.2.11") 
-- Performing Test ARCH_SUPPORTS_64BIT_ATOMICS
-- Performing Test ARCH_SUPPORTS_64BIT_ATOMICS - Success
-- Performing Test HAS_PTHREAD_SETAFFINITY_NP
-- Performing Test HAS_PTHREAD_SETAFFINITY_NP - Success
-- Performing Test LIBC_HAS_BACKTRACE
-- Performing Test LIBC_HAS_BACKTRACE - Success
-- Detecting USDT support
-- Detecting USDT support - not found
-- Checking for module 'libuv>=1.0.0'
--   Found libuv, version 1.34.0
-- Checking for module 'libwslay'
--   Found libwslay, version 1.1.0
-- Checking for module 'libbrotlidec'
--   Package 'libbrotlidec', required by 'virtual:world', not found
-- Checking for module 'libbrotlienc'
--   Package 'libbrotlienc', required by 'virtual:world', not found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/axel/h2o-master/build

$ make -C build
...
[ 29%] Linking C shared library libh2o-evloop.so
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: cannot find -llibbrotlidec
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: cannot find -llibbrotlienc
collect2: error: ld returned 1 exit status
...
```

Reply:
Update https://github.com/h2o/h2o/pull/2326/commits/3a88f159fbc35cf3f2caa33c0049b883d983143c

Reply:
@syohex `make` runs through when brotli lib not installed, `make check` build phase is failing however when brotli lib not installed.  When brotli is present, both make and make check are working.  
The solution really is the combination of your 1st approach to be applied when brotli is not found (= fall back to the embedded code) and the current approach of linking it in when brotli is present at the host.

```
$ make -C build check
...
[ 16%] Linking C shared library libh2o.so
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: cannot find -llibbrotlidec
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: cannot find -llibbrotlienc
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: cannot find -llibbrotlidec
/usr/lib/gcc/x86_64-alpine-linux-musl/9.2.0/../../../../x86_64-alpine-linux-musl/bin/ld: cannot find -llibbrotlienc
...
```

cmake invocation  

```
$ cmake -B build . -DCMAKE_BUILD_TYPE=None -DBUILD_SHARED_LIBS=ON -DCMAKE_INS
TALL_LIBDIR=/usr/lib -DCMAKE_INSTALL_PREFIX=/usr -DWITH_MRUBY=ON
-- The C compiler identification is GNU 9.2.0
-- The CXX compiler identification is GNU 9.2.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.6.3") 
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found OpenSSL: /usr/lib/libcrypto.so (found version "1.1.1g")  
-- Found ZLIB: /lib/libz.so (found version "1.2.11") 
-- Performing Test ARCH_SUPPORTS_64BIT_ATOMICS
-- Performing Test ARCH_SUPPORTS_64BIT_ATOMICS - Success
-- Performing Test HAS_PTHREAD_SETAFFINITY_NP
-- Performing Test HAS_PTHREAD_SETAFFINITY_NP - Success
-- Performing Test LIBC_HAS_BACKTRACE
-- Performing Test LIBC_HAS_BACKTRACE - Success
-- Detecting USDT support
-- Detecting USDT support - not found
-- Checking for module 'libuv>=1.0.0'
--   Found libuv, version 1.34.0
-- Checking for module 'libwslay'
--   Found libwslay, version 1.1.0
-- Checking for module 'libbrotlidec'
--   Package 'libbrotlidec', required by 'virtual:world', not found
-- Checking for module 'libbrotlienc'
--   Package 'libbrotlienc', required by 'virtual:world', not found
-- Configuring done
-- Generating done
-- Build files have been written to: /home/axel/h2o-master/build
``